File: src/models/services/BookingService.ts
import { ServiceBase } from "../../base/ServiceBase.js";
import { BookingModel } from "../request/BookingModel";
import { Response } from "../responses/Response";

export class BookingService extends ServiceBase {
  constructor() {
    super("/booking");
  }

  async getBookingIds<T>(
    params: URLSearchParams = new URLSearchParams(),
    config = this.defaultConfig,
  ): Promise<Response<T>> {
    config.params = params;
    return await this.get<T>(this.url, config);
  }

  async getBooking<T>(id: unknown, config = this.defaultConfig): Promise<Response<T>> {
    return await this.get<T>(`${this.url}/${id}`, config);
  }

  async addBooking<T>(
    booking: BookingModel | any,
    config = this.defaultConfig,
  ): Promise<Response<T>> {
    return await this.post(this.url, booking, config);
  }

  async updateBooking<T>(
    id: unknown,
    booking: BookingModel | any,
    config = this.defaultConfig,
  ): Promise<Response<T>> {
    return await this.put(`${this.url}/${id}`, booking, config);
  }

  async partialUpdateBooking<T>(
    id: unknown,
    booking: BookingModel | any,
    config = this.defaultConfig,
  ): Promise<Response<T>> {
    return await this.patch(`${this.url}/${id}`, booking, config);
  }

  async deleteBooking<T>(id: unknown, config = this.defaultConfig): Promise<Response<T>> {
    return await this.delete(`${this.url}/${id}`, config);
  }
}

File: src/models/services/BookingService.py
from src.base.ServiceBase import ServiceBase
from src.models.request.BookingModel import BookingModel
from src.models.responses.Response import Response

class BookingService(ServiceBase):
    def __init__(self):
        super().__init__("/booking")

    def get_booking_ids(self, params: dict = None, config: dict = None) -> Response:
        if config is None:
            config = self.default_config
        if params:
            config["params"] = params
        return self.get(self.url, config)

    def get_booking(self, booking_id: int, config: dict = None) -> Response:
        if config is None:
            config = self.default_config
        return self.get(f"{self.url}/{booking_id}", config)

    def add_booking(self, booking: BookingModel, config: dict = None) -> Response:
        if config is None:
            config = self.default_config
        return self.post(self.url, booking, config)

    def update_booking(self, booking_id: int, booking: BookingModel, config: dict = None) -> Response:
        if config is None:
            config = self.default_config
        return self.put(f"{self.url}/{booking_id}", booking, config)

    def partial_update_booking(self, booking_id: int, booking: BookingModel, config: dict = None) -> Response:
        if config is None:
            config = self.default_config
        return self.patch(f"{self.url}/{booking_id}", booking, config)

    def delete_booking(self, booking_id: int, config: dict = None) -> Response:
        if config is None:
            config = self.default_config
        return self.delete(f"{self.url}/{booking_id}", config)
